django_prices/models.py

```diff
@@ -149,3 +149,83 @@ def clean(self, value, model_instance):
         # is_relation = False, but we rely on our net and gross fields for
         # actual validation.
         return value
+
+
+class MoneyCurrencyField:
```

Comment:
Why not call it `MoneyField` (and `TaxedMoneyField`)?

Reply:
I'm not pretty sure I'm understand your question.
Are you asking why is not it a descriptor for `MoneyField`? The usage could be very straightforward in those approach, I think:
```
class CurrencyModel(models.Model):
    currency = models.CharField(max_length=3, default='BTC')
    money = MoneyField(currency_field='currency', max_digits=9, decimal_places=2)
    money_gross = MoneyField(currency_field='currency', max_digits=9, decimal_places=2)
    taxed_money = TaxedMoneyField(net_field='money', gross_field='money_gross')
```
I've had implemented it and I've had liked that code, but after discussion with @krzysztofwolski and his suggestion to keep it simple I've pushed the version that could be found in this PR.

Reply:
No, I'm asking about the weird naming scheme. I prefer it being a descriptor that internally uses two explicit fields, one for the amount and one for the currency.

Reply:
Sorry, I missed the point.
The reason is that replacing old `MoneyField` (with `models.DecimalField` as a base class) by the new one, descriptor-only implementation (with `object` as a base class) is not possible. Django migrations system cannot deal with it; moreover, past migrations would fail (https://docs.djangoproject.com/en/2.2/howto/custom-model-fields/#changing-a-custom-field-s-base-class).
But you're right, those names are not perfect. What do you think about moving that field to the separate module with, e.g., usage `from django_prices.models.multicurrency import MoneyField`?

Reply:
I would love to entirely drop the previous field and provide instruction on how to fix the existing migrations (replacing it with a properly configured DecimalField). We can make it a new major release if required.

`MoneyCurrencyField` is not only ugly but also incorrect as the field stores _money_ that consists of _amount_ and _currency_. So it's either a MoneyField or an AmoundAndCurrencyField (which is also ugly and does not indicate the actual type returned).